src/ziggurat/init.clj

```diff
@@ -143,12 +143,18 @@
                            (shutdown-agents))
             "Shutdown-handler")))
 
+(declare StreamRoute)
+
 (s/defschema StreamRoute
   (s/conditional
    #(and (seq %)
          (map? %))
-   {s/Keyword {:handler-fn (s/pred #(fn? %))
-               s/Keyword   (s/pred #(fn? %))}}))
+   {s/Keyword {:handler-fn                     (s/pred #(fn? %))
+               (s/optional-key :consumer-type) (s/enum :default :stream-joins)
+               (s/optional-key :input-topics)  (s/pred vector?)
```

Comment:
Why are we taking input-topics from the start-fn and not the config?

Reply:
I will move it to config.edn

Reply:
can we also put it here for testing purposes?

Reply:
done, I have placed it on the config.edn as a map